src/main/java/org/kohsuke/github/Requester.java

```diff
@@ -902,8 +904,17 @@ private void setRequestMethod(HttpURLConnection connection) throws IOException {
             // don't wrap exception in HttpException to preserve backward compatibility
             throw e;
         } catch (IOException e) {
-            if (e instanceof SocketTimeoutException && timeouts > 0) {
-                LOGGER.log(INFO, "timed out accessing " + uc.getURL() + "; will try " + timeouts + " more time(s)", e);
+            if (((e instanceof SocketException && e.getCause() instanceof SocketTimeoutException)
+                    || e instanceof SocketTimeoutException) && timeouts > 0) {
+                LOGGER.log(INFO,
+                        "timed out accessing  " + uc.getURL() + ". Sleeping 5 seconds before retrying... ; will try"
+                                + timeouts + " more time(s)",
+                        e);
+                try {
+                    Thread.sleep(5000);
```

Comment:
Make a configurable static for this and probably make it 500 millis to start? 
```suggestion
                    Thread.sleep(Requester.retryTimeoutMillis);
```